addons/website_membership/controllers/main.py

```diff
@@ -100,12 +117,21 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=0, *
         if free_partner_ids:
             memberships_data.append({'id': 'free', 'name': _('Free Members')})
             if not membership_id or membership_id == 'free':
-                memberships_partner_ids['free'] = free_partner_ids
+                if count_members < offset + limit:
+                    memberships_partner_ids['free'] = free_partner_ids[offset-count_members:offset+limit-count_members]
```

Comment:
I noticed that with having 4 silver members, and a bunch of free partners, on the first pages, only the 4 silvers appeared, and from page 2 you could see the free partners.

The first page should have contained the 4 silver members, then `limit - 4` free members, taking account of the offset.